samples/csharp/end-to-end-apps/Forecasting-Sales/README.md

```diff
@@ -0,0 +1,347 @@
+
+When learning/researching the samples, you can focus choose to focus on one of the algorithms/scenarios/models.
+
+![Build -> Train -> Evaluate -> Consume](docs/images/modelpipeline.png)
+
+#### 1. Load the Dataset
+
+Both the **Regression** and **Time Series** samples start by loading data using **TextLoader**. To use **TextLoader**, we must specify the type of the class that represents the data schema. Our class type is **ProductData**. 
+
+```csharp
+ public class ProductData
+    {
+        // The index of column in LoadColumn(int index) should be matched with the position of columns in the underlying data file.
+
+        // The next column is used by the Regression algorithm as the Label (e.g. the value that is being predicted by the Regression model).
+        [LoadColumn(0)]
+        public float next;
+
+        [LoadColumn(1)]
+        public string productId;
+
+        [LoadColumn(2)]
+        public float year;
+
+        [LoadColumn(3)]
+        public float month;
+
+        [LoadColumn(4)]
+        public float units;
+
+        [LoadColumn(5)]
+        public float avg;
+
+        [LoadColumn(6)]
+        public float count;
+
+        [LoadColumn(7)]
+        public float max;
+
+        [LoadColumn(8)]
+        public float min;
+
+        [LoadColumn(9)]
+        public float prev;
+    }
+```
+Load the dataset into the **DataView**. 
+
+```csharp
+
+var trainingDataView = mlContext.Data.LoadFromTextFile<ProductData>(dataPath, hasHeader: true, separatorChar:',');
+
+```
+
+In the following steps, we will build the pipeline transformations, specify which trainer/algorithm to use, evaluate the models, and test their predictions. This is where the steps start to differ between the [**Regression**](#2-regression-create-the-pipeline) and [**Time Series**](#6-time-series-create-the-pipeline) samples - the remainder of this walkthrough looks at each of these algorithms separately.
+
+
+#### 2. Regression: Create the Pipeline
+
+This step shows how to create the pipeline that will later be used for building and training the **Regression** model.
+
+Specifically, we do the following transformations:
+- Concatenate current features to a new column named **NumFeatures**.
+- Transform **productId** using [one-hot encoding](https://en.wikipedia.org/wiki/One-hot).
+- Concatenate all generated features in one column named **Features**.
+- Copy **next** column to rename it to **Label**.
+- Specify the **Fast Tree Tweedie** trainer as the algorithm to apply to the model.
+
+You can load the dataset either before or after designing the pipeline. Although this step is just configuration, it is lazy and won't be loaded until training the model in the next step.
+
+[Model build and train](./src/eShopForecastModelsTrainer/ProductModelHelper.cs)
```

Comment:
This link does not work. There is an src folder in the same directory as the readme (so it would be /src instead of ./src I believe), but I can't find the ProductModelHelper.cs in the eShopForecastModelsTrainer folder.

Reply:
resolved: 080839e5b1e63b57dccff55e980847cbff5efe59